### ISSUE TITLE
More issues with newly added events

### ISSUE THREAD

**Issue by leonstr** [mdjnelson/moodle-mod_customcert#570]
I've tested the fixes for  #564  and see the following issues:

1. Unexpected _template_updated_ on create template (but not add activity instance). It's unexpected because it's in addition to the expected _template_created_ event. 
2. No event on clicking **Save changes** or **Add element** on **customcert/edit.php**. The latter action implicitly saves template changes, e.g. any name change. I'd expect a _template_updated_.
3. Event descriptions when adding, updating and deleting a course activity instance show "created the certificate template", "updated the certificate template" and "deleted the certificate template". Whereas these should say "created the certificate", "updated the certificate" and "deleted the certificate" respectively, with "... template" only displayed when adding, updating and deleting templates under **Site administration** > **Plugins** > **Activity modules** > **Custom certificate** > **Custom certificate settings**, **Manage templates**).
4. Loading a template for a course activity instance includes a _page_created_ event with context: system, whereas surely this should be context: cmid.

Tested with 7ee063e (MOODLE_401_STABLE) on Moodle 4.1.3+ (build: 20230526).

**Comment by leonstr** [comment]
As a first go at fixing these I've [added branch 564-401 commit 1941274](https://github.com/mdjnelson/moodle-mod_customcert/compare/MOODLE_401_STABLE...leonstr:moodle-mod_customcert:564-401?expand=1).

Another issue: I don't like that every _page_updated_ also has a _template_updated_, if there's one action then two events seems wrong. Having one _template_updated_ + one or more _page_updated_ would make sense. I'd need to do a bit more testing to see if the _template_updated_ could be removed from **template->save_page()**, i.e. does a _template_created_ or _template_updated_ definitely get triggered every time **save_page()** is called? I haven't tried to fix this.

**Comment by leonstr** [comment]
Another issue: There are two _template_created_ events when copying a template. Possibly **template->copy_to_template()** doesn't need the:

```
                \mod_customcert\event\template_created::create_from_template($copytotemplate)->trigger();
```

because **manage_templates.php** calls **template->create()** which calls:

```
        \mod_customcert\event\template_created::create_from_template($template)->trigger();
```

?

**Comment by leonstr** [comment]
There's a further issue that rearranging elements by drag and drop then saving these this doesn't trigger any events. At this stage I think this specific issue should be added to it's own ticket, patching the above issues has taken long enough and retro-fitting events to `rearrange.php` will take a bit of thought.

**Comment by mdjnelson** [comment]
@leonstr i've lost track. Can you help me out and let me know if this can be closed?

**Comment by leonstr** [comment]
Yes, as far as I know the issues were addressed in 62b2c2f and this can be closed.
